src/solver_info.jl

```diff
@@ -4,18 +4,37 @@ export can_solve_mip, can_solve_socp, can_solve_sdp, can_solve_exp
 export set_default_solver, get_default_solver, isinstalled
 
 function set_default_solver(solver::MathProgBase.AbstractMathProgSolver)
+    Base.depwarn(
+        "Default solvers are deprecated and will not be provided in a future release.
+        You can still manually load and use solvers such as ECOS by running:
+        using ECOS
+        p = minimize(...);
+        solve!(p, ECOSSolver())",
```

Comment:
Perhaps use multi-line strings, since these are strings across multiple lines? e.g.
```julia
Base.depwarn("""
    Default solvers are deprecated and will not be provided in a future release.
    You can still manually load and use solvers such as ECOS by running:
        using ECOS
        p = minimize(...)
        solve!(p, ECOSSolver())
    """,
    :set_default_solver
)
```

Reply:
I was mostly trying to follow whatever style was being used below, but that does seem cleaner.

Reply:
```
error("The default solver is set to `nothing`
          You must have at least one solver installed to use Convex.
          You can install a solver such as SCS by running:
          Pkg.add(\"SCS\")
          You will have to restart Julia after that.")
```

Reply:
Opted to make an issue to cleanup the formatting more generally. https://github.com/JuliaOpt/Convex.jl/issues/251